src/meta/TableSpec.h

```diff
@@ -185,5 +202,52 @@ struct TableSpecEqual {
 
 using TableSpecSet = nebula::common::unordered_set<TableSpecPtr, TableSpecHash, TableSpecEqual>;
 
+constexpr auto HOUR_MINUTES = 60;
+constexpr auto MINUTE_SECONDS = 60;
+constexpr auto DAY_HOURS = 24;
+constexpr auto HOUR_SECONDS = HOUR_MINUTES * MINUTE_SECONDS;
+constexpr auto DAY_SECONDS = HOUR_SECONDS * DAY_HOURS;
+
+const static nebula::common::unordered_map<nebula::meta::PatternMacro, std::string> patternStr{ { nebula::meta::PatternMacro::DATE, "date" }, { nebula::meta::PatternMacro::HOUR, "hour" }, { nebula::meta::PatternMacro::MINUTE, "minute" }, { nebula::meta::PatternMacro::SECOND, "second" }, { nebula::meta::PatternMacro::TIMESTAMP, "timestamp" } };
+const static nebula::common::unordered_map<nebula::meta::PatternMacro, nebula::meta::PatternMacro> childPattern{ { nebula::meta::PatternMacro::DATE, nebula::meta::PatternMacro::HOUR }, { nebula::meta::PatternMacro::HOUR, nebula::meta::PatternMacro::MINUTE }, { nebula::meta::PatternMacro::MINUTE, nebula::meta::PatternMacro::SECOND } };
+const static nebula::common::unordered_map<nebula::meta::PatternMacro, int> partitionInSeconds{ { nebula::meta::PatternMacro::DATE, DAY_SECONDS }, { nebula::meta::PatternMacro::HOUR, HOUR_SECONDS }, { nebula::meta::PatternMacro::MINUTE, MINUTE_SECONDS } };
+const static nebula::common::unordered_map<nebula::meta::PatternMacro, int> partitionSize{ { nebula::meta::PatternMacro::DATE, DAY_HOURS }, { nebula::meta::PatternMacro::HOUR, HOUR_MINUTES }, { nebula::meta::PatternMacro::MINUTE, MINUTE_SECONDS } };
+
+// check if pattern string type
+inline nebula::meta::PatternMacro extractPatternMacro(std::string pattern) {
```

Comment:
const std::string&

Reply:
I think we should add a unit test for this function.

Reply:
agree